### PR TITLE
Fix UV install in Makefile

### PR DESCRIPTION
**Pull request by thomasjpfan**
This PR fixes the CI issue in `main`.

https://github.com/flyteorg/flytekit/actions/runs/9100354302/job/25015010454

### CODE REVIEW

**Comment by eapolinario** [comment]
why is this happening now? I see that CI checks in master are failing (the run you linked in the description).

File: .github/workflows/pythonbuild.yml [hunk 1]
```diff
@@ -259,9 +259,12 @@ jobs:
           uv pip freeze
       - name: Install FlyteCTL
         uses: unionai-oss/flytectl-setup-action@master
+        with:
+          # Remove version and go back to latest when unionai-oss/flytectl-setup-action is fixed
+          version: "0.8.18"
       - name: Setup Flyte Sandbox
         run: |
-          flytectl demo start
+          flytectl demo start --version=v0.8.18
```
Comment (eapolinario) [inline .github/workflows/pythonbuild.yml:267]: we don't need to specify the version here. This flag maps to a flyte release, not flytectl.

File: Makefile [hunk 2]
```diff
@@ -30,7 +30,7 @@ setup: install-piptools ## Install requirements
 # Warning: this will install the requirements in your system python
 .PHONY: setup-global-uv
 setup-global-uv:
-	uv pip install --system -r dev-requirements.in
+	SETUPTOOLS_SCM_PRETEND_VERSION="3.0.0.dev0" uv pip install --system -r dev-requirements.in
```
Comment (eapolinario) [inline Makefile:33]: I'm still confused. Why are we seeing this now and now when I introduced uv in https://github.com/flyteorg/flytekit/pull/2403?

File: .github/workflows/pythonbuild.yml [hunk 3]
```diff
@@ -259,9 +259,12 @@ jobs:
           uv pip freeze
       - name: Install FlyteCTL
         uses: unionai-oss/flytectl-setup-action@master
+        with:
+          # Remove version and go back to latest when unionai-oss/flytectl-setup-action is fixed
```
Comment (eapolinario) [inline .github/workflows/pythonbuild.yml:263]: https://github.com/unionai-oss/flytectl-setup-action/pull/5 is the fix.

File: Makefile (hunk 2 above)
Comment (austin362667) [reply Makefile:33]: I'm curious about the reason too.

And after adding `SETUPTOOLS_SCM_PRETEND_VERSION`, it indeed fixed my broken CI.

**Review by austin362667 — APPROVED** [review]

File: .github/workflows/pythonbuild.yml (hunk 3 above)
Comment (eapolinario) [reply .github/workflows/pythonbuild.yml:263]: You can revert this change. The above PR was merged and I tested it in https://github.com/flyteorg/flyte/pull/5377.

File: .github/workflows/pythonbuild.yml (hunk 1 above)
Comment (eapolinario) [reply .github/workflows/pythonbuild.yml:267]: ditto.

File: Makefile (hunk 2 above)
Comment (pingsutw) [reply Makefile:33]: feel like the regression in UV. they just released yesterday. https://pypi.org/project/uv/#history

**Review by austin362667 — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (eapolinario) [reply Makefile:34]: This is unfortunate. I couldn't find any issue on [their](https://github.com/astral-sh/uv/issues) tracker though.

**Review by eapolinario — APPROVED** [review]

**Review by eapolinario — DISMISSED** [review]

**Review by eapolinario — APPROVED** [review]
I'd still love to understand what change in `uv` to cause this, but let's unblock for now.

**Comment by eapolinario** [comment]
I finally got to the bottom of this. It started to happen when I moved `flytectl` to the monorepo. Here's the fix: https://github.com/flyteorg/flyte/pull/5419.

**Pull request merged**